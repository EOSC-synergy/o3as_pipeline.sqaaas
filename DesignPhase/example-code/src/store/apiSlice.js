// Import the RTK Query methods from the React-specific entry point
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'

// Define our single API slice object
export const apiSlice = createApi({
  // The cache reducer expects to be added at `state.api` (already default - this is optional)
  reducerPath: 'api',
  
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.o3as.fedcloud.eu/api/v1' }),
  
  // The "endpoints" represent operations and requests for this server
  endpoints: builder => ({
    
    getModels: builder.query({
      query: () => '/models'
    }),
    getPlots: builder.query({
      query: () => '/plots'
    }),
    getRawPlotData: builder.mutation({
      query: payload => ({
        // /data/tco3_return?begin=1959&end=2100&lat_min=-90&lat_max=90
        url: `/data/${payload.plotType}?begin=1959&end=2100&lat_min=${payload.lat_min}&lat_max=${payload.lat_max}`,
        method: "POST",
        body: payload.modelList,
      })
    }),
    
    getBuiltPlotData: builder.mutation({
      query: payload => ({
        
        url: `/plots/${payload.plotType}?begin=1959&end=2100&lat_min=${payload.lat_min}&lat_max=${payload.lat_max}`,
        method: "POST",
        body: payload.modelList,
      })
    }),
    
  })
})

// Export the auto-generated hook for the query endpoints
export const { 
  useGetPostsQuery,
  useGetPostQuery,
  useGetBuiltPlotDataMutation,
  useGetRawPlotDataMutation,
 } = apiSlice