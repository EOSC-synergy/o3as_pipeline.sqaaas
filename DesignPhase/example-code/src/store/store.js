import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice'
import modelsReducer from './modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})