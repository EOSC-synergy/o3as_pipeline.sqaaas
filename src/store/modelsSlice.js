import { createSlice } from "@reduxjs/toolkit";

const STATISTICAL_VALUES = ["mean", "median", "derivative", "percentile"];

const initialState = {
    // currently active plot
    plotId: "tco3_zm",
    settings: {
        "tco3_zm": { 
            // this objects holds key-value-pairs, the keys being the model-group 
            // identifier and the values being the settings for each group 
            "all": { 
                // model group storing all information until it is possible 
                // to implement more model groups
                name: "All OCTS models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        color: "#000000",
                        plotStyle: "solid",
                        isVisible: true, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        },
        "tco3_return": {
            "all": { 
                name: "All Return/Recovery models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        hidden: false, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                        color: "#000000",
                        plotStyle: "solid",
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        }
    }
}

const modelsSlice = createSlice({
    name: "models",
    initialState,
    reducers: {
        // set models
        // add models

        addModels(state, action) { 
            const { groupId, newModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            // Add model to modelList if it is not already included
            for (var i = 0; i < newModelList.length; i++) {
                if (activeModelList.indexOf(newModelList[i]) !== -1) {
                    continue;
                }

                activeModelList.push(newModelList[i]);
            }

            // ---------------------------------------------------------
            // TODO: Add Model information to the model lookup table
            // ---------------------------------------------------------


        }, // expects groupID, list of models 
        removeModels(state, action) { 
            const { groupId, removeModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            for (var i = 0; i < removeModelList.length; i++) {
                // Remove model from modelList
                activeModelList = activeModelList.filter(item => item !== removeModelList[i]);

                // Delete model object from models
                delete selectedGroup.models[removeModelList[i]];
            }

        }, // expects groupdID, list of models
        updatedModelGroup(state, action) { }, // implementing this requires more knowledge about the UI
        addedModelGroup(state, action) { },
        setVisibility(state, action) { 
            const { groupID, modelID, isVisible } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            } 
            activeModel.isVisible = isVisible;
        },
        setStatisticalValuesIncluded(state, action) { // this is for an individual model
            // svType has to be a string of the form: 
            // mean | derivative | median | percentile
            const { groupID, modelID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            }
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeModel[svType] = isIncluded;
        },
        setStatisticalValueForGroup(state, action) { // this is for an entire group
            const { groupID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeSettings[groupID].visibileSV[svType] = isIncluded;
        },
    }
})

export const { 
    addModels,
    removeModels,
    updatedModelGroup,
    addedModelGroup,
    setVisibility, 
    setStatisticalValuesIncluded, 
    setStatisticalValueForGroup
} = modelsSlice.actions;

export default modelsSlice.reducer;

export const selectCurrentModelGroups = state => state.models[state.plot.plotId];
export const selectCurrentModelGroup = (state, groupId) => state.models[state.plot.plotId][groupId];