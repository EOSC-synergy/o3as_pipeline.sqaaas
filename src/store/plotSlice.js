import { createSlice } from "@reduxjs/toolkit"

/**
 * The initial state of the plotSlice defines the data structure in the 
 * store. Each plot has its own settings i.e. title, location etc.
 */
const initialState = {
    
    plotId: "tco3_zm", // currently active plot
    // maps plotids to their settings
    settings: {
        "tco3_zm": {
            name: "OCTS", // should show up in the drop down menu
            title: "OCTS Plot", // the title shown in the apexcharts generated chart
            location: {
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 1960,
                maxX: 2100,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        },
        "tco3_return": {
            name: "Return/Recovery",
            title: "Return/Recovery Plot",
            location: { // custom user defined region
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 0,
                maxX: 0,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        }
    }
}

/**
 * The plotSlice is generated by the redux toolkit. The reducers are defined here
 * and the corresponding actions are auto-generated.
 */
const plotSlice = createSlice({
    name: "plot",
    initialState,
    reducers: {
        setActivePlotId(state, action) { // e.g. dispatch(setActivePlotId({id: "tco3_zm"}))
            const { id } = action.payload
            state.plotId = id
        }, 
        setTitle(state, action) { // e.g. dispatch(setActivePlotId({title: "OCTS Plot"}))
            const { title } = action.payload
            state.settings[state.plotId].title = title
        },
        setLocation(state, action) { 
            const {minLat, maxLat} = action.payload
            const location = state.settings[state.plotId].location
            location.minLat = minLat
            location.maxLat = maxLat
        },
        setDisplayXRange(state, action) { 
            const {minX, maxX} = action.payload
            const displayXRange = state.settings[state.plotId].displayXRange
            displayXRange.minX = minX
            displayXRange.maxX = maxX
        },
        setDisplayYRange(state, action) { 
            const {minY, maxY} = action.payload
            const displayYRange = state.settings[state.plotId].displayYRange
            displayYRange.minY = minY
            displayYRange.maxY = maxY
        },
        setMonths(state, action) { 
            state.settings[state.plotId].months = action.payload.months
        },
    }   
})


/**
 * The here listed actions are exported and serve as an interface for
 * the view (our react components).
 */
export const {
    setActivePlotId,
    setTitle,
    setLocation,
    setDisplayXRange,
    setDisplayYRange,
    setMonths,
} = plotSlice.actions

/**
 * The reducer combining all reducers defined in the plot slice. 
 * This has to be included in the redux store, otherwise dispatching 
 * the above defined actions wouldn't trigger state updates.
 */
export default plotSlice.reducer

/**
 * This selectors allows components to select the current plot id
 * from the store. The plot id is a string using the same naming as the
 * o3as api e.g. tco3_zm or tco3_return
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot id
 */
export const selectPlotId = state => state.plot.plotId

/**
 * This selectors allows components to select the current plot name
 * from the store. The plot name is the humand readable version specifying 
 * the plot e.g. OCTS or Return/Recovery
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot name
 */
export const selectPlotName = state => state.plot.settings[state.plot.plotId].name

/**
 * This selectors allows components to select the current plot title
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot title
 */
export const selectPlotTitle = state => state.plot.settings[state.plot.plotId].title

/**
 * This selectors allows components to select the current plot location
 * from the store. The location is an object containg a minLat and maxLat attribute.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current location that includes a minLat and maxLat attribute.  
 */
export const selectPlotLocation = state => state.plot.settings[state.plot.plotId].location

/**
 * This selectors allows components to select the current x range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minX and maxX
 */
export const selectPlotXRange = state => state.plot.settings[state.plot.plotId].displayXRange

/**
 * This selectors allows components to select the current y range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minY and maxY
 */

export const selectPlotYRange = state => state.plot.settings[state.plot.plotId].displayYRange

/**
 * This selectors allows components to select the current selected months
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {array} array of strings describing the current selected months
 */

export const selectPlotMonths = state => state.plot.settings[state.plot.plotId].months
