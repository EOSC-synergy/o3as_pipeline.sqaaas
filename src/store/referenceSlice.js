import { createSlice } from "@reduxjs/toolkit"

const initialState = {
    "OCTS": {
        year: 1980,
        model: "CCMI-1_ACCESS_ACCESS-CCM-refC2",
        visibile: false,
        offsetApplied: false,
    }
}

const referenceSlice = createSlice({
    name: "reference",
    initialState,
    reducers: {
        setReferenceYear(state, action) {
            const {year} = action.payload;
            state.OCTS.year = year;
         },
        setReferenceModel(state, action) { 
            const {model} = action.payload;
            state.OCTS.model = model;
        },
        setVisibile(state, action) { },
        setOffsetApplied(state, action) { },
    }
})

export const {
    setReferenceYear,
    setReferenceModel,
    setVisibile,
    setOffsetApplied,
} = referenceSlice.actions

export default referenceSlice.reducer

export const selectCurrentReferenceSettings = state=> state.reference[state.plot.plotId]
