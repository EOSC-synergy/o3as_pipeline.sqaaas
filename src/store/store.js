import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice/plotSlice'
import modelsReducer from './modelsSlice/modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})