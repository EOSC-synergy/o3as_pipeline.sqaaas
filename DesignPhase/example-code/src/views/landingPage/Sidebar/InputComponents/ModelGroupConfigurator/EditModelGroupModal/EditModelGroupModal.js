import React, { useState } from "react";
import { useDispatch } from "react-redux"
import { updatedModelGroup } from "../../../../../../store/modelsSlice"

/**
 * opens a modal where the user can edit an existing model group
 * @param {*} props 
 *  props.onClose -> function to call if modal should be closed
 *  props.open -> boolean whether the modal should be visible 
 *  props.error -> error function
 *  props.models -> models in this model group (also whether they are incoorperated into the mean/median/derivative calculation)
 *  props.name -> name of the model group
 *  props.id -> id of the model group
 * @returns a jsx containing a modal with a data grid with all models from the model group
 */
function EditModelGroupModal(props) {

    const dispatch = useDispatch()

    let i = props.onClose;
    i = props.open;
    i = props.error;

    i = props.models;
    i = props.name;
    i = props.id;

    i = props.editModelGroup;

    const handleOnSave = () => {
        dispatch(updatedModelGroup(someData))
    }

    return (
        <>
        </>
    );
}

export default EditModelGroupModal;