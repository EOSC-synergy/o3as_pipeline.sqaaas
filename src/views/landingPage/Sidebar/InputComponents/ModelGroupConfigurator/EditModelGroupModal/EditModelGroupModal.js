import React, { useState } from "react";
import { useDispatch } from "react-redux"
import { updatedModelGroup } from "../../../../../../store/modelsSlice"
import { Modal, Card, Button, Grid, Checkbox } from "@mui/material";
import { DataGrid } from '@mui/x-data-grid';
import models from "./models"
import SearchBar from "../SearchBar/SearchBar";
import { styled } from '@mui/material/styles';
import IconButton from '@mui/material/IconButton';
import DeleteIcon from '@mui/icons-material/Delete';
import AddIcon from '@mui/icons-material/Add';
import DoneIcon from '@mui/icons-material/Done';
import CancelIcon from '@mui/icons-material/Cancel';
import ClearIcon from '@mui/icons-material/Clear';

import { Typography } from "@mui/material";

const StyledDataGrid = styled(DataGrid)(({theme}) => ({
    height: "65%",
    marginTop: "3%",
}));

const rows = [];
const regex= /([a-z]|[A-Z]|[0-9]|-)*/g;

for(let i = 0; i < models.length; i++) {
    const model = models[i];
    const info = model.match(regex);
    rows.push({
        'model': info[0],
        'institute': info[2],
        'dataset + model': info[4],
        'id': i,
        'include in median': false,
        'include in mean': false,
        'include in percentile': false,
        'include in derivative': false,
        'visible': false
    })
}

/**
 * opens a modal where the user can edit an existing model group
 * @param {Object} props 
 * @param {function} props.onClose -> function to call if modal should be closed
 * @param {boolean} props.isOpen -> boolean whether the modal should be visible 
 * @param {function} props.reportError -> error function
 * @param {int} props.modelGroupId -> id of the model group
 * @returns a jsx containing a modal with a data grid with all models from the model group
 */
function EditModelGroupModal(props) {

    // const dispatch = useDispatch()

    let i = props.onClose;
    i = props.isOpen;
    i = props.reportError;
    i = props.modelGroupId;

    const [filteredRows, setFilteredRows] = React.useState(rows);

    // dispatch(updatedModelGroup(someData))

    const cardStyle = {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        width: '65%',
        height: '75%',
        bgcolor: "#FFFFFF",
        boxShadow: 24,
        p: 4,
    };

    const handleMedianChecked = () => {

    }

    const handleMeanChecked = () => {

    }

    const handleDerivativeChecked = () => {

    }

    const handlePercentileChecked = () => {

    }

    const handleVilibleChecked = () => {

    }

    
    //const makeRepeated = (arr, repeats) => Array.from({ length: repeats }, () => arr).flat();

    //rows = makeRepeated(rows, 10);

    const columns = [
        { field: 'id', headerName: 'ID', width: 90, hide: true },
        { field: 'model', headerName: 'Model', width: 120 },
        {
          field: 'institute',
          headerName: 'Institute',
          width: 150,
          editable: true,
        },
        {
          field: 'dataset + model',
          headerName: 'Dataset and Model',
          width: 225,
          editable: true,
        },
        {
            field: 'include in median',
            headerName: 'Median',
            sortable: false,
            width: 120,
            disableClickEventBubbling: true,
            renderCell: (params) => {
                return (
                    <div className="d-flex justify-content-between align-items-center" style={{ cursor: "pointer" }}>
                        <Checkbox onClick={handleMedianChecked}/>
                    </div>
                );
             }
          },
          {
            field: 'include in mean',
            headerName: 'Mean',
            sortable: false,
            width: 120,
            disableClickEventBubbling: true,
            renderCell: (params) => {
                return (
                    <div className="d-flex justify-content-between align-items-center" style={{ cursor: "pointer" }}>
                        <Checkbox onClick={handleMeanChecked}/>
                    </div>
                );
             }
          },
          {
            field: 'include in derivative',
            headerName: 'Derivative',
            sortable: false,
            width: 120,
            disableClickEventBubbling: true,
            renderCell: (params) => {
                return (
                    <div className="d-flex justify-content-between align-items-center" style={{ cursor: "pointer" }}>
                        <Checkbox onClick={handleDerivativeChecked}/>
                    </div>
                );
             }
          },
          {
            field: 'include in percentile',
            headerName: 'Percentile',
            width: 120,
            sortable: false,
            disableClickEventBubbling: true,
            renderCell: (params) => {
                return (
                    <div className="d-flex justify-content-between align-items-center" style={{ cursor: "pointer" }}>
                        <Checkbox onClick={handlePercentileChecked}/>
                    </div>
                );
             }
          },
          {
            field: 'visible',
            headerName: 'Visible',
            sortable: false,
            width: 120,
            disableClickEventBubbling: true,
            renderCell: (params) => {
                return (
                    <div className="d-flex justify-content-between align-items-center" style={{ cursor: "pointer" }}>
                        <Checkbox onClick={handleVilibleChecked} />
                    </div>
                );
             }
          },
          {
            field: 'delete',
            headerName: 'Delete',
            sortable: false,
            width: 120,
            disableClickEventBubbling: true,
            renderCell: (params) => {
                return (
                    <div className="d-flex justify-content-between align-items-center" style={{ cursor: "pointer" }}>
                        <IconButton aria-label="delete" color={"error"}>
                            <DeleteIcon />
                        </IconButton>
                    </div>
                );
             }
          },

    ];

    const foundIndices = (indexArray) => {
        setFilteredRows(indexArray.map(idx => rows[idx])); // translates indices into selected rows
    }

    return (
        <>
        {props.isOpen && 

            <Modal 
                open={props.isOpen}
                onClose={props.onClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <Card sx={cardStyle}>
                    <Grid container alignItems="flex-end" justifyContent="right" style={{marginBottom: "1%"}}>
                        
                        <IconButton aria-label="delete" color={"success"} size="large">
                            <DoneIcon fontSize="large"/>
                        </IconButton>
                        <IconButton onClick={props.onClose} aria-label="delete" color={"error"} size="large">
                            <ClearIcon fontSize="large"/>
                        </IconButton>
                    </Grid>
                    <div style={{width: "95%"}}>
                        <SearchBar 
                            inputArray={rows}
                            foundIndicesCallback={foundIndices}
                        />
                    </div>
                    <StyledDataGrid 
                            rows={filteredRows}
                            columns={columns}
                            pageSize={10}
                            rowsPerPageOptions={[5]}
                            // hideFooter
                            // autoHeight
                            disableSelectionOnClick
                            disableColumnMenu
                            disableColumnSelector
                    />
                    
                    <Grid container alignItems="center" justifyContent="center">
                        <IconButton aria-label="delete" color={"success"} size="large">
                            <AddIcon fontSize="large"/>
                        </IconButton>
                    </Grid>
                </Card>

            </Modal>
        }
        </>
    );
}

export default EditModelGroupModal;