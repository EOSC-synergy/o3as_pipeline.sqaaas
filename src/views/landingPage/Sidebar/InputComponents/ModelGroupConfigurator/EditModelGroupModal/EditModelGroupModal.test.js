import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux'
import '@testing-library/jest-dom';
import EditModelGroupModal from './EditModelGroupModal';
import { createTestStore } from "../../../../../../store/store"

let store
beforeEach(() => {
    store = createTestStore();
})

describe("test EditModelGroupModal rendering", () => {

    it("renders without crashing", () => {
        render(<Provider store={store}><EditModelGroupModal isOpen={true} onClose={() => {}} modelGroupId={0}/></Provider>);
    });

    it("renders correctly when open", () => {
        const { container } = render(<Provider store={store}><EditModelGroupModal isOpen={true} onClose={() => {}} modelGroupId={0}/></Provider>);
        expect(container).toMatchSnapshot();
        expect(container).toBeVisible();
    });

    it("closes the modal correctly when changes are discarded", () => {
        const onClose = jest.fn()
        const { container, getByTestId } = render(<Provider store={store}><EditModelGroupModal isOpen={true} onClose={onClose} modelGroupId={0}/></Provider>);
        const closeButton = getByTestId(/DiscardButton/);
        fireEvent.click(closeButton);
        expect(onClose).toHaveBeenCalled();
    });

    it("closes the modal correctly when changes are applied", () => {
        const onClose = jest.fn()
        const { container, getByTestId } = render(<Provider store={store}><EditModelGroupModal isOpen={true} onClose={onClose} modelGroupId={0}/></Provider>);
        const closeButton = getByTestId(/ApplyButton/);
        fireEvent.click(closeButton);
        expect(onClose).toHaveBeenCalled();
    });

    it("searches for a specific string", () => {
        const onClose = jest.fn()
        const { container, getByTestId, getByDisplayValue } = render(<Provider store={store}><EditModelGroupModal isOpen={true} onClose={onClose} modelGroupId={0}/></Provider>);
        const input = getByTestId("SearchbarInput");
        fireEvent.change(input, {target: {value: "AAA"}})
    });

    it("closes the modal correctly when changes are applied", () => {
        const onClose = jest.fn()
        const { container, getByTestId } = render(<Provider store={store}><EditModelGroupModal isOpen={true} onClose={onClose} modelGroupId={0}/></Provider>);
        //const closeButton = getByTestId(/Row0TypeMean/);
        //fireEvent.click(closeButton);
        //expect(onClose).toHaveBeenCalled();
    });


});
