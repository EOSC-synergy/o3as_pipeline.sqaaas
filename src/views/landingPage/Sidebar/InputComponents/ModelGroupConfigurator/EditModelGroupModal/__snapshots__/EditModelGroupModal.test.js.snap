// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test EditModelGroupModal rendering renders correctly when open 1`] = `
<div
  aria-hidden="true"
/>
`;
